iceoryx_hoofs/include/iceoryx_hoofs/error_handling/error_handling.hpp

```diff
@@ -276,9 +242,44 @@ class ErrorHandler
 /// errorHandler(Error::kTEST__ASSERT_CALLED);
 /// ASSERT_TRUE(called);
 /// @endcode
+/// @tparam[in] Error type of the enum which is used to report the error
+/// @todo #590 use enable_if with is_enum
```

Comment:
Lets assume some user would like to transmit additional information like a state or stack trace inside a struct. With enforcing enums here we have to write more code while restricting the user unnecessarily.

Iceoryx does not have a problem in supporting any kind of type here so maybe we should leave it this way.